packages/@ember/-internals/glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -1514,62 +1486,6 @@ moduleFor(
       );
     }
 
-    ['@test this.attrs.foo === attrs.foo === @foo === foo']() {
```

Comment:
Perhaps this test should become `['@test @foo === this.foo']() {`? I'm not sure that is covered elsewhere, and it seems worth retaining that assertion.

Reply:
added back and modified test

Reply:
This should also still include `this.attrs.foo === this.foo === @foo`, the only thing removed is the "magical" rewrite of `{{attrs.foo}}` -> `{{@foo}}`.

Reply:
based on the attrs transform test, it seems like `this.attrs` is also deprecated:

```js
expectDeprecation(() => {
        this.assertTransformed(`{{this.attrs.foo}}`, `{{@foo}}`);
      }, /Using {{attrs}} to reference named arguments has been deprecated. {{attrs.foo}} should be updated to {{@foo}}./);
```
is this not the case?

Reply:
I think at that point the question becomes "Was `this.attrs` in `EmberComponent` deprecated? Can it be removed?`

If the `EmberComponent` system is setting a JS property `attrs` on the instance still, then `{{this.attrs.foo}}` remains a reasonable thing to use *even if it was deprecated*.

However I agree that the main goal here was the complete elimination of `this.attrs` in all places. I'll add this to our agenda for Friday with Ember framework core in case we don't have clarity by then.

Reply:
This was clarified with Ember FW Core on July 23rd.

Despite the fact that `this.attrs` was deprecated, we should retain support for it in 4.0. We *should* add an ember-template-lint rule against using `this.attrs` and `this.args` in templates, but as they are each properties on the backing class for a template there is no reason the syntax/usage itself is invalid. The framework should not do anything special.

So, yes, here we can retain `this.attrs.foo === this.foo === @foo` without deprecation.

Reply:
k made those changes